src/components/App.tsx

```diff
@@ -2394,12 +2395,16 @@ class App extends React.Component<AppProps, AppState> {
     if (lastPointerDownHittingLinkIcon && LastPointerUpHittingLinkIcon) {
       const url = this.hitLinkElement?.link;
       if (url) {
-        const target = isLocalLink(url) ? "_self" : "_blank";
-        const newWindow = window.open(undefined, target);
-        // https://mathiasbynens.github.io/rel-noopener/
-        if (newWindow) {
-          newWindow.opener = null;
-          newWindow.location = normalizeLink(url);
+        if (this.props.onLinkOpen) {
+          this.props.onLinkOpen(url, event);
```

Comment:
should we also pass the `element` ? might be useful for the host

Reply:
passing the element is a good suggestion

Reply:
Following on the idea to pass the element... It would be also great to have an `onHoverLink` prop as well. I already provide link previews in Obsidian, I will definitely want to provide the same for element links.

https://user-images.githubusercontent.com/14358394/152410535-5556ebc7-7741-4573-b0ec-50974849a54b.mp4

Reply:
Definitely, will create a separate PR for this